multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/mmrcms/MMRCMS.kt

```diff
@@ -1,529 +1,487 @@
-import kotlinx.serialization.json.jsonPrimitive
-import okhttp3.OkHttpClient
+import okhttp3.FormBody
+import okhttp3.HttpUrl.Companion.toHttpUrl
 import okhttp3.Request
 import okhttp3.Response
+import org.jsoup.nodes.Document
 import org.jsoup.nodes.Element
 import rx.Observable
+import rx.Single
+import rx.schedulers.Schedulers
 import uy.kohesive.injekt.injectLazy
-import java.text.ParseException
 import java.text.SimpleDateFormat
 import java.util.Locale
-import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantLock
 
 abstract class MMRCMS(
     override val name: String,
     override val baseUrl: String,
-    override val lang: String,
-    sourceInfo: String = "",
-) : HttpSource() {
-    open val jsonData = if (sourceInfo == "") {
-        SourceData.giveMetaData(baseUrl)
-    } else {
-        sourceInfo
-    }
+    final override val lang: String,
+    private val dateFormat: SimpleDateFormat = SimpleDateFormat("d MMM. yyyy", Locale.US),
+) : ParsedHttpSource() {
+
+    override val supportsLatest = true
+
+    override fun headersBuilder() = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+
+    protected open val json: Json by injectLazy()
 
     /**
-     * Parse a List of JSON sources into a list of `MyMangaReaderCMSSource`s
-     *
-     * Example JSON :
-     * ```
-     *     {
-     *         "language": "en",
-     *         "name": "Example manga reader",
-     *         "base_url": "https://example.com",
-     *         "supports_latest": true,
-     *         "item_url": "https://example.com/manga/",
-     *         "categories": [
-     *             {"id": "stuff", "name": "Stuff"},
-     *             {"id": "test", "name": "Test"}
-     *         ],
-     *         "tags": [
-     *             {"id": "action", "name": "Action"},
-     *             {"id": "adventure", "name": "Adventure"}
-     *         ]
-     *     }
-     *
-     *
-     * Sources that do not supports tags may use `null` instead of a list of json objects
-     *
-     * @param sourceString The List of JSON strings 1 entry = one source
-     * @return The list of parsed sources
-     *
-     * isNSFW, language, name and base_url are no longer needed as that is handled by multisrc
-     * supports_latest, item_url, categories and tags are still needed
-     *
-     *
+     * The path used in the URL for the manga pages. Can be
+     * changed if needed as some sites modify it to other words.
      */
-    private val json: Json by injectLazy()
-    val jsonObject = json.decodeFromString<JsonObject>(jsonData)
-    override val supportsLatest = jsonObject["supports_latest"]!!.jsonPrimitive.boolean
-    open val itemUrl = jsonObject["item_url"]!!.jsonPrimitive.content
-    open val categoryMappings = mapToPairs(jsonObject["categories"]!!.jsonArray)
-    open var tagMappings = jsonObject["tags"]?.jsonArray?.let { mapToPairs(it) } ?: emptyList()
+    protected open val itemPath = "manga"
```

Comment:
I'm not a fan of adding a bunch of `open val`s (i.e. the current approach of Madara and MangaThemesia). These configurations are hard to discover and less performant. How about moving them into the constructor like the date format?

Reply:
I'm not a fan of having too many things inside a constructor/function call, unless there's a way to "enforce" passing named arguments a la Python's `kwargs`.

Reply:
I think you mean the `/` and `*`. I think you can hack it in Kotlin by inserting a class with private constructor lol.

Reply:
> I'm not a fan of having too many things inside a constructor/function call

Think data class constructors and Composable functions.